### PR TITLE
Custom file name length

### PR DESCRIPTION
**Pull request by ferchinas**
Change the hard-coded maximum character size in the filename from 11 to a constant variable

### CODE REVIEW

**Comment by charlesbaynham** [comment]
Hello, 

Thank you for the PR, and for helping to improve OSFS!

The 8+3 length was originally chosen to conform to the old DOS format: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/18e63b13-ba43-4f5f-a5b7-11e871b71f14. That was an arbitrary choice, but I don't really want to encourage people to change it: the presence of an OSFS header at the start of your memory is currently a promise that it'll conform to the format as specified. If people start changing filename sizes, that's no longer true.

However, I don't mind using a const variable in the code so that power users can alter it if they really want to: I'd just rather not advertise this in the readme.

I'd be happy to accept this PR with those tweaks! I'll leave a couple of comments on the code too.

File: README.md [hunk 1]
```diff
@@ -89,6 +89,7 @@ Internals
 This library has no support for fragmented files or directories. File names
 are in 8.3 format: 8 chars followed by 3 for an extension. Filenames will be
 padded to 11 chars by spaces.
+You can change the filename size by modifying the fileName_length variable from 11 to the number of characters you need.
```
Comment (charlesbaynham) [inline README.md]: As mentioned elsewhere, I'd rather not change the readme and keep this a purely internal change.

File: src/OSFS.h [hunk 2]
```diff
@@ -42,6 +42,8 @@
 #include <Arduino.h>
 
 namespace OSFS {
+	//File name lengths
+	constexpr uint8_t fileName_length = 11;
```
Comment (charlesbaynham) [inline src/OSFS.h]: The convention in the rest of the code is UPPERCASE_NAMES for constant variables. Could you change to this? Also, I think a `size_t` is a more appropriate type here.

**Review by charlesbaynham — CHANGES_REQUESTED** [review]
A stylistic tweak and change to the readme.

**Comment by ferchinas** [comment]
Hi Charles,
Thank you very much for your library, I have been playing with it these days integrating it with JChristensen / extEEPROM and bblanchon / ArduinoJson to store json files in my eeprom memory.

I apologize if I have not made the pull request flow correctly, it is the second I have done in my life.
Please tell me if I did something wrong.

All the best

**Comment by charlesbaynham** [comment]
You're very welcome! I hope it helps you out. 

The PR looks good to me: I'm just tweaking it to put the default filename length back to 11 chars otherwise it would break people's existing installations.

**Pull request merged**